### ISSUE TITLE
Edge error on selection when defaultEdges are connected to Redux store

### ISSUE THREAD

**Issue by alessandropisu** [wbkd/react-flow#1993]
### Describe the Bug

If the user pass to _DefaultEdges_ prop a variable containing a portion of store state of Redux and an edge selection operation is performed, **Cannot add property selected, object is not extensible** error is shown. There are some sort of shallow copy of value passed to DefaultEdges that causes an immutability error.
This doesn't happen with _DefaultNodes_ and with zustand instead Redux as state manager.

### Your Example Website or App

https://codesandbox.io/s/sandpack-project-forked-c03e86?file=/App.js

### Steps to Reproduce the Bug or Issue

Click on the edge or select multiple nodes with "Shift" selection

### Expected behavior

No errors should be displayed, or not allow the user to pass a direct reference of Redux state as prop.

### Screenshots or Videos

![Schermata 2022-03-21 alle 15 22 18](https://user-images.githubusercontent.com/38405118/159281368-971b718d-920a-48ac-a079-da5f7ef10cdd.png)

### Platform

- OS: macOS
- Browser: Chrome
- Version: v10
- State manager: Redux (v7.2.6) + Redux Toolkit (v.1.8.0)

**Comment by chrtze** [comment]
Hi @alessandropisu! I think this might be a problem with using ImmerJS in Redux Toolkit. Internally, we are not cloning the edge objects, so it might be that you are passing immutable ImmerJS objects into React Flow. This might not be a problem with nodes as we are parsing them into a slightly different data structure internally. Does this help?

**Comment by alessandropisu** [comment]
Hello @chrtze, thanks for your answer, the biggest doubt was about the error only during edges selection and not on nodes (I forgot they have different structures internally in ReactFlow).

For those who have the same problem and are reading: just [deep clone](https://flaviocopes.com/how-to-clone-javascript-object/) your edges array before pass them to DefaultEdges prop :)

**Comment by moklick** [comment]
That's a good solution for now @alessandropisu . Thanks for sharing :)

**Comment by nkalinov** [comment]
Guys I just stumbled upon the same issue and while deep-copying the edges fixes the issue itself I have another weird behavior - The first time when I try to drag a multi-selected group my selection is automatically unselected. Then after the second time it works.
I'm using Recoil, controlled RF props (edges and nodes), and then right before I render RF I have something like this:
```
  // Fix multi-selection of edges - https://github.com/wbkd/react-flow/issues/1993
  const edges = useMemo(() => JSON.parse(JSON.stringify(diagram.edges)), [diagram.edges]);

  return <ReactFlow nodes={diagram.nodes} edges={edges} ... />
```
Here's a video preview but if it's not obvious I can also try to create a Codepen repro:
https://user-images.githubusercontent.com/2777825/173307516-57389899-0532-47ac-ba46-2be0bfdc9aa4.mov

Any ideas @moklick ?

**Comment by kahboom** [comment]
@nkalinov did you ever solve this? do you have any example using React Flow with Recoil?

**Comment by moklick** [comment]
This error should be fixed in the latest version (I think that we fixed it in v10.3.0)
